Added gql integration docs (#7946)
getsentry/sentry-docs: src/platforms/python/common/configuration/integrations/gql.mdx

---

> title: gql
> description: gql GraphQL client integration

Reviewer: ```suggestion
title: GQL
description: GQL GraphQL client integration
```

sidebar_order: 55
---

> The [gql](https://gql.readthedocs.io/en/stable/) integration allows you to monitor all GraphQL requests that your Python application makes using the `gql` library. Specifically, this integration reports every [`TransportQueryError`](https://gql.readthedocs.io/en/latest/modules/transport_exceptions.html#gql.transport.exceptions.TransportQueryError) to Sentry and provides additional context including the URL the request was made to, GraphQL query which caused the error, and the `errors` information that the GraphQL server returns.

Reviewer: ```suggestion
The [GQL](https://gql.readthedocs.io/en/stable/) integration allows you to monitor all GraphQL requests that your Python application makes using the `gql` library. Specifically, this integration reports every [`TransportQueryError`](https://gql.readthedocs.io/en/latest/modules/transport_exceptions.html#gql.transport.exceptions.TransportQueryError) to Sentry and provides additional context including:
- The URL the request was made to
- The GraphQL query which caused the error
- The `errors` information that the GraphQL server returns
```

Reviewer: suggest capitalization fix and breaking up features into a bulleted list for readability

## Install

Make sure you have the latest version of the Sentry SDK installed:

```bash
pip install --upgrade sentry-sdk
```

## Configure

To enable the integration, add `GQLIntegration()` to your `integrations` list, like so:

<SignInNote />

```python
import sentry_sdk
from sentry_sdk.integrations.gql import GQLIntegration

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[
        GQLIntegration(),
    ],
    # Uncomment the following line to send GraphQL
    # queries and error information to Sentry:
    #
    # send_default_pii=True,
)
```

> ### Data Privacy & The `gql` Integration

Reviewer: This seems like it should be its own H2 heading

```suggestion
## Data Privacy and the GQL Integration
```

> To ensure data privacy, the [SDK avoids sending any information which may potentially contain personally identifiable information (PII)](/platforms/python/configuration/options/#send-default-pii) by default.

Reviewer: ```suggestion
To ensure data privacy, the SDK avoids sending any information that may potentially contain personally identifiable information (PII) by default. See the docs for the [`send_default_pii` flag](/platforms/python/configuration/options/#send-default-pii) for more information.
```

> For the `gql` integration, this policy means that the SDK **refrains from sending the GraphQL queries and the `errors` information** that the GraphQL server returns unless you override the default configuration via the `send_default_pii` flag.

Reviewer: ```suggestion
For the GQL integration, this policy means that the SDK **doesn't send** the GraphQL queries and the `errors` information that the GraphQL server returns, unless you override the default configuration via the `send_default_pii` flag.
```

<Alert level="warning">

>   Prior to enabling the <code>send_default_pii</code> flag, please ensure that
>   you have configured Sentry's{" "}
>   <a href="/platforms/python/data-management/sensitive-data/">
>     sensitive data scrubbing
>   </a>{" "}
>   tools properly to scrub all PII.

Reviewer: I believe you can use normal markdown if you put leading & trailing newlines in the alert

```suggestion

Before enabling the `send_default_pii` flag, ensure you've configured Sentry's [sensitive data scrubbing](/platforms/python/data-management/sensitive-data/) tools to scrub all PII.
  
```

</Alert>

> If you would like Sentry to record the GraphQL queries and the `errors` information returned by the GraphQL server, you can uncomment the `send_default_pii=True` line in the code snippet above.

Reviewer: ```suggestion
To have Sentry record the GraphQL queries and the `errors` information returned by the GraphQL server, uncomment the `send_default_pii=True` line in the code snippet above.
```
